.github/workflows/delegator_generic_slsa3.yml

```
@@ -0,0 +1,257 @@
+# Copyright 2022 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA builder delegator
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  SLSA_OUTPUTS_DIR: __SLSA_OUTPUTS_DIR__
+  SLSA_ARTIFACTS_FILE: artifacts-layout.json
+  SLSA_PREDICATE_FILE: predicate.json
+
+on:
+  workflow_call:
+    # Secrets provided as arguments are passed as inputs to the callback Action.
+    secrets:
+      secret1:
+        description: >
+          Optional secret.
+
+          This argument is passed, unchanged, to the builder's Action's `build-artifacts`.
+        required: false
+      secret2:
+        description: >
+          Optional secret.
+
+          This argument is passed, unchanged, to the builder's Action's `build-artifacts`.
+        required: false
+
+    # Inputs provided as arguments are passed as inputs to the callback Action,
+    # formatted as a map.
+    inputs:
+      slsa-token:
+        description: "The SLSA token identifying the request"
```

```suggestion
        description: "The signed SLSA token identifying the request"
```

we should remember to document carefully the format of the signed SLSA token input: that it's the `<slsa-token>.<signature-bundle>` (or reversed, i forgot)

---

is this relevant for public use? the TRW writers only need to know they should provide the slsa token generated with slsa-setup Action, nothing more, I think.

+1 on documenting. How about a README.md under the `setup-token` Action? Maybe we can also rename it to `delegator/init`

---

I added a README.md under the setup-token.

---

> the TRW writers only need to know they should provide the slsa token generated with slsa-setup Action, nothing more, I think.

README.md under setup-token is good, I think it may be relevant for them if they're curious to inspect the inputs and outputs they're passing in the TRW (like Fredrik had asked at some point).